src/components/JaegerIntegration/JaegerResults/SpanTable.tsx

```diff
@@ -72,8 +72,8 @@ export class SpanTableC extends React.Component<SpanDetailProps, SpanDetailState
       node = sp.tags.filter(tag => tag.key === 'node_id')[0].value;
     }
     const srv = sp.process.serviceName.split('.')[0];
-    const regex = new RegExp(`${srv}-v[0-9]*`);
-    const result = node !== '' ? regex.exec(node) : null;
+    const regex = new RegExp(`([a-z0-9-\\.]+)-[a-z0-9]+-[a-z0-9]+\\.[a-z0-9-]+`);
+    const result = node !== '' ? regex.exec(node.split('~')[2]) : null;
```

Comment:
in the following lines 79/81/82, it's `result[1]` that must be used and not `result[0]`, else we get the full name (like `ai-visitors-6b86864b-bn6fh.default`) instead of just the extracted desired part (like `ai-visitors`)

Reply:
mmm yes...now this change ...but Is the workload `ai-visitors-6b86864b-bn6fh` in youre environment and the route '/namespaces/deafult/workloads/ai-visitors-6b86864b-bn6fh?tab=logs' ? In that case we must remove the .default but we need to check in the expression if there are multiples `.<word>`.

Let me work on this

Reply:
the regex you wrote is fine. There's the parenthesis that allow to extract the first part. You just need to use result[1], I've tested it, it works fine.

Reply:
PS my workload is simply `ai-visitors`

Reply:
yes so here is the problem because result[1] will be ai-visitors-6b86864b-bn6fh.default and the link to workloads should be /namespaces/deafult/workloads/ai-visitors?tab=logs . Am I right?

Reply:
no, `result[0]` is `ai-visitors-6b86864b-bn6fh.default` and `result[1]` is `ai-visitors`, so `result[1]` is fine